### PR TITLE
Add testing and model validation

### PR DESCRIPTION
**Pull request by byiringiroscar**
####  Add testing and model validation
This PR is raised to implement the following features 🆗 :

- [x] Add Model validation
- [x] Test model 
- [x] Setup linters

### How should it be tested ⬇️ ⬇️ ⬇️ 

clone this repository
     
```
git clone https://github.com/byiringiroscar/wikipedia-project-with-rails
```
Navigate to folder
```
cd wikipedia-project-with-rails
```
Switch to this feature/validate-test
```
git switch feature/validate-test
```
Run project and Installation
- open the command prompt and run :
```
bundle install
```
```
rspec
```

### Thanks for reviewing my PR 😄

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+  it 'is valid with valid attributes' do
+    expect(subject).to be_valid
+  end
+
+  it 'is not valid without a name' do
+    subject.name = nil
+    expect(subject).to_not be_valid
+  end
+end
```
Comment (AnselemOdims) [inline spec/models/user_spec.rb:37]: - [ ] Good job working on this, but please include tests for the test case `PostsCounter must be an integer greater than or equal to zero.`

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  it 'is not valid when comments_counter is string' do
+    subject.comments_counter = 'string'
+    expect(subject).to_not be_valid
+  end
+
+  it 'is not valid when likes_counter is string' do
+    subject.likes_counter = 'string'
+    expect(subject).to_not be_valid
+  end
+end
```
Comment (AnselemOdims) [inline spec/models/post_spec.rb]: - [ ] Good job working on this, but please kindly include test for the following test cases
  - Title must not exceed 250 characters.
  - CommentsCounter must be greater than or equal to zero.
  - LikesCounter must be greater than or equal to zero.

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @byiringiroscar , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by arslanbisharat** [comment]
Hi @byiringiroscar,
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- There is no linter error is present. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
